src/main/java/org/folio/rest/impl/OrdersImpl.java

```diff
@@ -367,5 +383,17 @@ public void getOrdersReceivingHistory(int offset, int limit, String query, Strin
 
   }
 
+    private CompletableFuture<Errors> validate(String lang, Map<String, String> okapiHeaders, Context vertxContext, HttpClientInterface httpClient, CompositePurchaseOrder compPO) {
```

Comment:
This method can be simplified a little via the use of `thenCompose(...)`

Reply:
Thanks, Craig! Good point. Since the validation of vendor and access providers was conceived by me as independent actions, I can suggest using thenCombine() method instead. This will allow us to perform validations in parallel mode, and I think this will have a positive impact on performance. And I removed misleading indentation.

Reply:
Yes, this is better... you can simplify even futher:

```
  private CompletableFuture<Errors> validate(String lang, Map<String, String> okapiHeaders, Context vertxContext,
      HttpClientInterface httpClient, CompositePurchaseOrder compPO) {
    VendorHelper vendorHelper = new VendorHelper(lang, httpClient, okapiHeaders, vertxContext);
    return vendorHelper.validateVendor(compPO)
      .thenCombine(vendorHelper.validateAccessProviders(compPO), (errors, accessProvidersErrors) -> {
        errors.getErrors().addAll(accessProvidersErrors.getErrors());
        errors.setTotalRecords(errors.getErrors().size());
        return errors;
      });
  }
```

Reply:
Done.